homeassistant/components/melcloud/config_flow.py

```
@@ -0,0 +1,96 @@
+        for entry in self._async_current_entries():
+            if entry.data.get(CONF_EMAIL, entry.title) == email:
+                entry.connection_class = self.CONNECTION_CLASS
+                self.hass.config_entries.async_update_entry(
+                    entry, data={CONF_EMAIL: email, CONF_TOKEN: token}
+                )
+                return self.async_abort(
+                    reason="already_configured",
+                    description_placeholders={"email": email},
+                )
+
+        return self.async_create_entry(
+            title=email, data={CONF_EMAIL: email, CONF_TOKEN: token}
+        )
+
+    async def _init_client(self, email: str, password: str) -> pymelcloud.Client:
+        return await pymelcloud.login(
+            email, password, self.hass.helpers.aiohttp_client.async_get_clientsession(),
+        )
+
+    async def _init_client_with_token(self, token: str) -> pymelcloud.Client:
+        return pymelcloud.Client(
+            token, self.hass.helpers.aiohttp_client.async_get_clientsession(),
+        )
+
+    async def _create_client(
+        self, email: str, client_creator: Callable[[], pymelcloud.Client],
+    ):
+        """Create client."""
+        try:
+            client = await client_creator()
+            with timeout(10):
+                await client.update_confs()
+        except asyncio.TimeoutError:
+            return self.async_abort(reason="cannot_connect")
+        except ClientResponseError as err:
+            if err.status == 401 or err.status == 403:
+                return self.async_abort(reason="invalid_auth")
+            return self.async_abort(reason="cannot_connect")
+        except ClientError:
+            _LOGGER.exception("ClientError")
+            return self.async_abort(reason="cannot_connect")
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected error creating device")
+            return self.async_abort(reason="unknown")
```

Do not catch the bare exception, but let it bubble up.

---

Done